diff.py

```diff
@@ -1046,6 +1071,113 @@ def search_map_file(
     return None, None
 
 
+def parse_elf_data_references(data: bytes) -> List[Tuple[int, int, str]]:
+    e_ident = data[:16]
+    if e_ident[:4] != b"\x7FELF":
+        return []
+
+    SHT_SYMTAB = 2
+    SHT_REL = 9
+    SHT_RELA = 4
+
+    is_32bit = e_ident[4] == 1
+    is_little_endian = e_ident[5] == 1
+    str_end = "<" if is_little_endian else ">"
+    str_off = "I" if is_32bit else "Q"
+    sym_size = {"B": 1, "H": 2, "I": 4, "Q": 8}
+
+    def read(spec: str, offset: int) -> Tuple[int, ...]:
+        spec = spec.replace("P", str_off)
+        size = sum(sym_size[c] for c in spec)
```

Comment:
You can use `struct.calcsize(...)` instead. (This would also let you use things like "5H" if you wanted)